plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -579,16 +595,19 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
     }
 
-    protected static String likeToRegexp(String like)
+    protected static String likeToRegexp(String like, String escape)
     {
+        checkState(escape.length() == 1, "Escape character must be a single char");
         // TODO: This can be done more efficiently by using a state machine and iterating over characters (See io.trino.type.LikeFunctions.likePattern(String, char, boolean))
         String regexp = like.replaceAll(Pattern.quote("\\"), Matcher.quoteReplacement("\\\\")); // first, escape regexp's escape character
         for (char c : REGEXP_RESERVED_CHARACTERS) {
             regexp = regexp.replaceAll(Pattern.quote(String.valueOf(c)), Matcher.quoteReplacement("\\" + c));
         }
         return regexp
-                .replaceAll("%", ".*")
-                .replaceAll("_", ".");
+                .replaceAll("(?<!" + Pattern.quote(escape) + ")%", ".*")
```

Comment:
This is not correct. The escape may be itself escaped.

let's add a BCT test for that as it's seems easy to miss


BTW See above for a TODO

> `// TODO: This can be done more efficiently by using a state machine and iterating over characters (See io.trino.type.LikeFunctions.likePattern(String, char, boolean))`

it's not only more efficient, but -- with escapes -- is also simpler to implement.

Reply:
Done :) PTAL now